### ISSUE TITLE
Add support for setting on or off (not toggle)

### ISSUE THREAD

**Issue by hsaturn** [powerman/vim-plugin-AnsiEsc#21]
Hello

I have a script that uses AnsiEsc to reveal colors inside a QuickFix window (after make)

So I throw AnsiEsc then colors are on. Fine
When building a second time, I use the same map command that throws AnsiEsc => colors are now off.

The workaround is to check if the QuickFix list is on or off, But this could not be possible when using another plugin that could evolve, close QuickFix before reopening it etc...

The clean solution should be to have a pair of functions AnsiEscOn AnsiEscOff or whatever,(AnsiEscEnable(bool) ...

Something in autoload:
``` vim
fun! AnsiEsc#AnsiEscOn()
  let bn=bufnr("%")
  
  if !exists("s:AnsiEsc_enabled_{bn}") || s:AnsiEsc_enabled_{bn}==0
   call AnsiEsc#AnsiEsc(0)
  endif
endfun

fun! AnsiEsc#AnsiEscOff()
  let bn= bufnr("%")
  if exists("s:AnsiEsc_enabled_{bn}") && s:AnsiEsc_enabled_{bn}
   call AnsiEsc#AnsiEsc(0)
  endif
endfun
```
And the public interface in plugin:

``` vim
com! -bang -nargs=0 AnsiEscOn	:call AnsiEsc#AnsiEscOn()
com! -bang -nargs=0 AnsiEscOff	:call AnsiEsc#AnsiEscOff()
```

Best regards

**Comment by powerman** [comment]
Sounds like a good idea! Can you open a PR?

> When building a second time, I use the same map command that throws AnsiEsc => colors are now off.

Probably there is a way to change your command to call AnsiEsc only when QuickFix buffer is created for a first build, and then use already implemented `AnsiEsc#BufReadPost` when it's updated.

**Comment by hsaturn** [comment]
No, just because I cannot know if this is the 1st time because I use a plugin that use or re-use the QuickFix window. So the window is always there when mine is called. And I cannot call AnsiEsc only once, because more than one QuickList could be opened, or even closed then re-opened later.

My point is : one cannot know if other plugin or user or whoever / whatever has already called AnsiEsc AND how many times AND on which windows etc.
It should be programmatically possible to be deterministic on the state of AnsiEsc after a call (outside the plugin).
-> Having the state of AnsiEsc should be enough, but this state is private.

So, I'll stick to my AnsiEscOn/Off patch. I'll probably have to fork if you cannot / do not want to use this.

Best Regards

**Comment by powerman** [comment]
> you cannot / do not want to use this

You probably misunderstood me.
As I've said - it's a good idea to have this feature, so if you'll open a PR (using current implementation, maybe with extra documentation update related to new feature) I'll be happy to merge.

My note about possibility to implement what you need without this feature is just an idea and not a reason to block the feature.

**Comment by hsaturn** [comment]
Sorry, you're right: I've read the lower part of your answer while the upper part was clear enough, even for a non english like me :-)

I did not start directly by a PR because I thought my (quick made) code could be wrong.
I'll write a PR.

Best regards

Regards.

**Comment by nkh** [comment]
@powerman  @hsaturn 

Hi guy, any progress on this?

**Comment by powerman** [comment]
Nope. PRs are welcome.

**Comment by nkh** [comment]
unfortunately my vimscript-fu is zero!

I did notice that, at line 34, call AnsiEsc#AnsiEsc(0) was called twice, was that on purpose?

**Comment by powerman** [comment]
I think so. It's triggered from line 2268 when handling BufReadPost to re-initialize after buffer changes.

**Comment by powerman** [comment]
After all, I'm just willing to maintain the plugin and provide easier way to install and collaborate on it, but I'm not an author of original plugin, so I'm not really familiar with all of it's code.

**Comment by hsaturn** [comment]
Sorry. I use this patch since months now. With no problems. I Forgot thé pr

**Comment by nkh** [comment]
@hsaturn please PR! so I don't have to :)

**Comment by wwalker** [comment]
:+1: 

I'll try to create a PR.
